src/codec/flate/encoder.rs

```diff
@@ -83,8 +83,9 @@ impl Encode for FlateEncoder {
             }
         }
 
-        self.flushed = true;
-        Ok(!output.unwritten().is_empty())
+        let internal_flushed = !output.unwritten().is_empty();
+        self.flushed = internal_flushed;
```

Comment:
I think maybe it's better to check if the length of `output.unwritten()` has been changed?

There could be some data flushed to the buffer, but not enough to make it full, and maybe there could be morr data left to flush?

Not sure if this can happen, cc @robjtede wdyt?

Reply:
> The loop right above stops only when output.written() does not change anymore,

Thanks! I missed that when thinking about this


> If the buffer does not ever have enough space to encode anything with FlushCompress::Sync, then I don't know what happens.

I think we should return `Ok(false)` and prompt user to do another flush?

Reply:
Ok so I read the code again, I notice that it would only set it as flushed, when there are some bytes left unwritten in the buffer.

I think I was wrong about "what if the buffer was fully written and still left some data to flush", as the code would set flushed to false in that situation.